### PR TITLE
[PW-2739] Handle Error and Rejected result codes

### PR DESCRIPTION
**Pull request by cyattilakiss**
## Summary
Add PaymentResponseHandler to CardsPaymentMethodHandler

## Tested scenarios
Rejected card payment

### CODE REVIEW

File: src/Handlers/CardsPaymentMethodHandler.php [hunk 1]
```diff
@@ -401,7 +414,7 @@ public function preparePaymentsRequest(
             ),
             $transaction->getOrder()->getOrderNumber(),
             $this->configurationService->getMerchantAccount(),
-            $transaction->getReturnUrl(),
+            'test url',
```
Comment (msilvagarcia) [inline src/Handlers/CardsPaymentMethodHandler.php:417]: Expected?

File: src/Handlers/PaymentResponseHandler.php [hunk 2]
```diff
@@ -26,14 +26,18 @@
...
 use Symfony\Component\HttpFoundation\JsonResponse;
+use Shopware\Core\Checkout\Payment\Cart\AsyncPaymentTransactionStruct;
+use Shopware\Core\Checkout\Payment\Exception\AsyncPaymentProcessException;
+use Shopware\Core\System\SalesChannel\SalesChannelContext;
+use Symfony\Component\HttpFoundation\RedirectResponse;
 
 class PaymentResponseHandler
 {
-    const ADYEN_MERCHANT_REFERENCE = 'adyenMerchantReference';
+    const ADYEN_MERCHANT_REFERENCE = 'merchantReference';
```
Comment (acampos1916) [inline src/Handlers/PaymentResponseHandler.php:40]: This must match the GET var the is returned in the URL, in PS is adyenMerchantReference. Did you change it for some different reason?

Comment (cyattilakiss) [reply src/Handlers/PaymentResponseHandler.php:40]: aaaa sorry so this is for the GET parameter - I use it for the synch response merchantReference var
I will create a new one!

File: src/Handlers/CardsPaymentMethodHandler.php (hunk 1 above)
Comment (cyattilakiss) [reply src/Handlers/CardsPaymentMethodHandler.php:417]: Thanks for noticing, I will put back the original one and if it's necessary we can adjust it later to a new url

**Pull request merged**